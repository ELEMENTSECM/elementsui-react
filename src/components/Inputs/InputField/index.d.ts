import * as React from 'react';

export interface InputFieldProps {
	/**
	 * Input field placeholder text
	 */
	placeholder?: string;
	/**
	 * Input field is marked with red asterisk as required if set to true
	 */
	required?: boolean;
	/**
	 * Input field is disabled is set to true
	 */
	disabled?: boolean;
	/**
	 * Error message shown under the input field
	 */
	errorMessage?: string;
}

export const InputField: React.SFC<InputFieldProps>;
