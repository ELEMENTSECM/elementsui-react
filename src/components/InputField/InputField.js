import * as React from 'react';
import { TextField } from 'office-ui-fabric-react/lib/TextField';
import PropTypes from 'prop-types';

function InputField({ placeholder, required, disabled, errorMessage }) {
	return (
		<TextField
			required={required}
			disabled={disabled}
			errorMessage={errorMessage}
			placeholder={placeholder}
		/>
	);
}

InputField.propTypes = {
	/** Do you want a fancy placeholder? */
	placeholder: PropTypes.string,
	required: PropTypes.bool,
	disabled: PropTypes.bool,
	errorMessage: PropTypes.string
};

export default InputField;
