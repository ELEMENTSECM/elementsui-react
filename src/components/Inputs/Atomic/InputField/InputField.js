import * as React from 'react';
import { TextField } from 'office-ui-fabric-react/lib/TextField';
import PropTypes from 'prop-types';

function InputField({ placeholder, required, disabled, errorMessage }) {
	return (
		<TextField
			required={required}
			disabled={disabled}
			errorMessage={errorMessage}
			placeholder={placeholder}
		/>
	);
}

InputField.propTypes = {
	/** Input field placeholder text */
	placeholder: PropTypes.string,
	/** Input field is marked with red asterisk as required if set to true */
	required: PropTypes.bool,
	/** Input field is disabled is set to true */
	disabled: PropTypes.bool,
	/** Error message shown under the input field */
	errorMessage: PropTypes.string
};

InputField.defaultProps = {
	disabled: false,
	required: false
};

export default InputField;
