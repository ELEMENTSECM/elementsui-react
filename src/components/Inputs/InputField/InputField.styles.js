export const styles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: [
			'ms-InputField',
			{
				// place your styles here
			},
			className
		]
	};
};
