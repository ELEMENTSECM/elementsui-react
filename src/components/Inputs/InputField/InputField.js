import * as React from 'react';
import PropTypes from 'prop-types';
import { TextField } from 'office-ui-fabric-react/lib/TextField';
import { getStyles } from './InputField.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';

export function InputField(props) {
	const { placeholder, required, disabled, errorMessage, className, theme, getStyles } = props;
	const classNames = classNamesFunction()(getStyles, props);
	return (
		<TextField
			className={classNames.root}
			required={required}
			disabled={disabled}
			errorMessage={errorMessage}
			placeholder={placeholder}
		/>
	);
}

InputField.propTypes = {
	/** Input field placeholder text */
	placeholder: PropTypes.string,
	/** Input field is marked with red asterisk as required if set to true */
	required: PropTypes.bool,
	/** Input field is disabled is set to true */
	disabled: PropTypes.bool,
	/** Error message shown under the input field */
	errorMessage: PropTypes.string
};

InputField.defaultProps = {
	disabled: false,
	required: false
};

export default styled(customizable('InputField', ['theme'])(InputField), getStyles);
