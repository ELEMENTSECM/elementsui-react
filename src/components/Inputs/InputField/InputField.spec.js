import React from 'react';
import renderer from 'react-test-renderer';
import InputField from './InputField';

describe('InputField', () => {
	test('displays correctly', () => {
		const tree = renderer
			.create(<InputField id="testInputField" label="test field" placeholder="Search..." />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
