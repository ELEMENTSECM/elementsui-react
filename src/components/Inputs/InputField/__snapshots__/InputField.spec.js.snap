// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField displays correctly 1`] = `
<div
  className="ms-TextField root_df60a678 ms-InputField root-40"
>
  <div
    className="ms-TextField-wrapper wrapper_df60a678"
  >
    <div
      className="ms-TextField-fieldGroup fieldGroup_df60a678"
    >
      <input
        aria-invalid={false}
        className="ms-TextField-field field_df60a678"
        disabled={false}
        id="testInputField"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        placeholder="Search..."
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
